src/iuse.cpp

```diff
@@ -4733,6 +4733,118 @@ int iuse::throwable_extinguisher_act(player *, item *it, bool, const tripoint &p
     it->active = false;
     return 0;
 }
+int iuse::capture_monster_act(player *p, item *it, bool, const tripoint& pos){
+  if(it->has_var("contained_name")){
+    try{
+        tripoint target;
+        if(g->is_empty(pos))
+        {
+            target=pos;//it's been activated somewhere where there isn't a player or monster, good.=
+        }else{
+            if(it->has_flag("PLACE_RANDOMLY")){
+                std::vector<tripoint> valid;
+                for(int x=p->posx()-1;x<=p->posx()+1;x++){
+                    for(int y=p->posy()-1;y<=p->posy()+1;y++){
+                        tripoint dest(x,y,pos.z);
+                        if(g->is_empty(dest)){
+                            valid.push_back(dest);
+                        }
+                    }
+                }
+                target=random_entry(valid);
```

Comment:
If no entry is valid, this will be a `tripoint( 0, 0, 0 )`, which isn't a good spot for a critter.